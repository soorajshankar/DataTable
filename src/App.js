import React from "react";
import logo from "./logo.svg";
import "./App.css";
import Table from "./Table";

function App() {
  return <Table data={getData(5000)} />;
}
const getData = size => {
  let data = [];
  for (let i = 0; i <= size; i++) {
    data.push({ id: i, name: `${i} Guy` });
  }
  return data;
};

export default App;
