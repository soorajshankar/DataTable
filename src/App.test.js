import React from "react";
import { render } from "@testing-library/react";
import App from "./App";
it("renders welcome message", () => {
  const wrapper = render(<App />);
  // console.log(wrapper.find("table").debug());
  expect(wrapper).toMatchSnapshot();
});
