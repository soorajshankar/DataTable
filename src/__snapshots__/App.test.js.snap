// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders welcome message 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        id="root"
        style="padding: 10px; height: 100vh;"
      >
        <div
          style="height: 50%; overflow: hidden;"
        >
          <h2>
            With 500000 data points
          </h2>
          <div
            class="vir-table"
          >
            <div
              class="vir-table-header"
            >
              <table>
                <colgroup>
                  <col
                    style="background-color: rgb(250, 250, 250); width: 20px; min-width: 20px;"
                  />
                  <col
                    style="background-color: rgb(250, 250, 250); width: 100px; min-width: 100px;"
                  />
                  <col
                    style="background-color: rgb(250, 250, 250); width: 10%; min-width: 10%;"
                  />
                  <col
                    style="background-color: rgb(250, 250, 250); width: 10%; min-width: 10%;"
                  />
                </colgroup>
                <thead>
                  <th>
                    <input
                      type="checkbox"
                    />
                  </th>
                  <th
                    align="right"
                  >
                    ID
                  </th>
                  <th>
                    NAME
                  </th>
                  <th
                    align="right"
                  >
                    IID
                  </th>
                </thead>
              </table>
            </div>
            <div
              class="vir-table-body"
              style="max-height: 100%; overflow: auto;"
            >
              <table>
                <colgroup>
                  <col
                    style="background-color: rgb(255, 255, 255); width: 20px; min-width: 20px;"
                  />
                  <col
                    style="background-color: rgb(255, 255, 255); width: 100px; min-width: 100px;"
                  />
                  <col
                    style="background-color: rgb(255, 255, 255); width: 10%; min-width: 10%;"
                  />
                  <col
                    style="background-color: rgb(255, 255, 255); width: 10%; min-width: 10%;"
                  />
                </colgroup>
                <tbody>
                  <tr>
                    <td>
                      <input
                        type="checkbox"
                      />
                    </td>
                    <td
                      align="right"
                    >
                      0
                    </td>
                    <td>
                      0 Guy
                    </td>
                    <td
                      align="right"
                    >
                      0
                    </td>
                  </tr>
                  <tr>
                    <td>
                      <input
                        type="checkbox"
                      />
                    </td>
                    <td
                      align="right"
                    >
                      1
                    </td>
                    <td>
                      1 Guy
                    </td>
                    <td
                      align="right"
                    >
                      1
                    </td>
                  </tr>
                  <tr>
                    <td>
                      <input
                        type="checkbox"
                      />
                    </td>
                    <td
                      align="right"
                    >
                      2
                    </td>
                    <td>
                      2 Guy
                    </td>
                    <td
                      align="right"
                    >
                      2
                    </td>
                  </tr>
                  <tr>
                    <td>
                      <input
                        type="checkbox"
                      />
                    </td>
                    <td
                      align="right"
                    >
                      3
                    </td>
                    <td>
                      3 Guy
                    </td>
                    <td
                      align="right"
                    >
                      3
                    </td>
                  </tr>
                  <tr>
                    <td>
                      <input
                        type="checkbox"
                      />
                    </td>
                    <td
                      align="right"
                    >
                      4
                    </td>
                    <td>
                      4 Guy
                    </td>
                    <td
                      align="right"
                    >
                      4
                    </td>
                  </tr>
                  <tr>
                    <td>
                      <input
                        type="checkbox"
                      />
                    </td>
                    <td
                      align="right"
                    >
                      5
                    </td>
                    <td>
                      5 Guy
                    </td>
                    <td
                      align="right"
                    >
                      5
                    </td>
                  </tr>
                  <tr>
                    <td>
                      <input
                        type="checkbox"
                      />
                    </td>
                    <td
                      align="right"
                    >
                      6
                    </td>
                    <td>
                      6 Guy
                    </td>
                    <td
                      align="right"
                    >
                      6
                    </td>
                  </tr>
                  <tr>
                    <td>
                      <input
                        type="checkbox"
                      />
                    </td>
                    <td
                      align="right"
                    >
                      7
                    </td>
                    <td>
                      7 Guy
                    </td>
                    <td
                      align="right"
                    >
                      7
                    </td>
                  </tr>
                  <tr>
                    <td>
                      <input
                        type="checkbox"
                      />
                    </td>
                    <td
                      align="right"
                    >
                      8
                    </td>
                    <td>
                      8 Guy
                    </td>
                    <td
                      align="right"
                    >
                      8
                    </td>
                  </tr>
                  <tr>
                    <td>
                      <input
                        type="checkbox"
                      />
                    </td>
                    <td
                      align="right"
                    >
                      9
                    </td>
                    <td>
                      9 Guy
                    </td>
                    <td
                      align="right"
                    >
                      9
                    </td>
                  </tr>
                  <tr>
                    <td>
                      <input
                        type="checkbox"
                      />
                    </td>
                    <td
                      align="right"
                    >
                      10
                    </td>
                    <td>
                      10 Guy
                    </td>
                    <td
                      align="right"
                    >
                      10
                    </td>
                  </tr>
                  <tr>
                    <td>
                      <input
                        type="checkbox"
                      />
                    </td>
                    <td
                      align="right"
                    >
                      11
                    </td>
                    <td>
                      11 Guy
                    </td>
                    <td
                      align="right"
                    >
                      11
                    </td>
                  </tr>
                  <tr>
                    <td>
                      <input
                        type="checkbox"
                      />
                    </td>
                    <td
                      align="right"
                    >
                      12
                    </td>
                    <td>
                      12 Guy
                    </td>
                    <td
                      align="right"
                    >
                      12
                    </td>
                  </tr>
                  <tr>
                    <td>
                      <input
                        type="checkbox"
                      />
                    </td>
                    <td
                      align="right"
                    >
                      13
                    </td>
                    <td>
                      13 Guy
                    </td>
                    <td
                      align="right"
                    >
                      13
                    </td>
                  </tr>
                  <tr>
                    <td>
                      <input
                        type="checkbox"
                      />
                    </td>
                    <td
                      align="right"
                    >
                      14
                    </td>
                    <td>
                      14 Guy
                    </td>
                    <td
                      align="right"
                    >
                      14
                    </td>
                  </tr>
                  <tr>
                    <td>
                      <input
                        type="checkbox"
                      />
                    </td>
                    <td
                      align="right"
                    >
                      15
                    </td>
                    <td>
                      15 Guy
                    </td>
                    <td
                      align="right"
                    >
                      15
                    </td>
                  </tr>
                  <tr>
                    <td>
                      <input
                        type="checkbox"
                      />
                    </td>
                    <td
                      align="right"
                    >
                      16
                    </td>
                    <td>
                      16 Guy
                    </td>
                    <td
                      align="right"
                    >
                      16
                    </td>
                  </tr>
                  <tr>
                    <td>
                      <input
                        type="checkbox"
                      />
                    </td>
                    <td
                      align="right"
                    >
                      17
                    </td>
                    <td>
                      17 Guy
                    </td>
                    <td
                      align="right"
                    >
                      17
                    </td>
                  </tr>
                  <tr>
                    <td>
                      <input
                        type="checkbox"
                      />
                    </td>
                    <td
                      align="right"
                    >
                      18
                    </td>
                    <td>
                      18 Guy
                    </td>
                    <td
                      align="right"
                    >
                      18
                    </td>
                  </tr>
                  <tr>
                    <td>
                      <input
                        type="checkbox"
                      />
                    </td>
                    <td
                      align="right"
                    >
                      19
                    </td>
                    <td>
                      19 Guy
                    </td>
                    <td
                      align="right"
                    >
                      19
                    </td>
                  </tr>
                  <tr>
                    <td>
                      <input
                        type="checkbox"
                      />
                    </td>
                    <td
                      align="right"
                    >
                      20
                    </td>
                    <td>
                      20 Guy
                    </td>
                    <td
                      align="right"
                    >
                      20
                    </td>
                  </tr>
                  <tr>
                    <td>
                      <input
                        type="checkbox"
                      />
                    </td>
                    <td
                      align="right"
                    >
                      21
                    </td>
                    <td>
                      21 Guy
                    </td>
                    <td
                      align="right"
                    >
                      21
                    </td>
                  </tr>
                  <tr>
                    <td>
                      <input
                        type="checkbox"
                      />
                    </td>
                    <td
                      align="right"
                    >
                      22
                    </td>
                    <td>
                      22 Guy
                    </td>
                    <td
                      align="right"
                    >
                      22
                    </td>
                  </tr>
                  <tr>
                    <td>
                      <input
                        type="checkbox"
                      />
                    </td>
                    <td
                      align="right"
                    >
                      23
                    </td>
                    <td>
                      23 Guy
                    </td>
                    <td
                      align="right"
                    >
                      23
                    </td>
                  </tr>
                  <tr>
                    <td>
                      <input
                        type="checkbox"
                      />
                    </td>
                    <td
                      align="right"
                    >
                      24
                    </td>
                    <td>
                      24 Guy
                    </td>
                    <td
                      align="right"
                    >
                      24
                    </td>
                  </tr>
                  <tr>
                    <td>
                      <input
                        type="checkbox"
                      />
                    </td>
                    <td
                      align="right"
                    >
                      25
                    </td>
                    <td>
                      25 Guy
                    </td>
                    <td
                      align="right"
                    >
                      25
                    </td>
                  </tr>
                  <tr>
                    <td>
                      <input
                        type="checkbox"
                      />
                    </td>
                    <td
                      align="right"
                    >
                      26
                    </td>
                    <td>
                      26 Guy
                    </td>
                    <td
                      align="right"
                    >
                      26
                    </td>
                  </tr>
                  <tr>
                    <td>
                      <input
                        type="checkbox"
                      />
                    </td>
                    <td
                      align="right"
                    >
                      27
                    </td>
                    <td>
                      27 Guy
                    </td>
                    <td
                      align="right"
                    >
                      27
                    </td>
                  </tr>
                  <tr>
                    <td>
                      <input
                        type="checkbox"
                      />
                    </td>
                    <td
                      align="right"
                    >
                      28
                    </td>
                    <td>
                      28 Guy
                    </td>
                    <td
                      align="right"
                    >
                      28
                    </td>
                  </tr>
                  <tr>
                    <td>
                      <input
                        type="checkbox"
                      />
                    </td>
                    <td
                      align="right"
                    >
                      29
                    </td>
                    <td>
                      29 Guy
                    </td>
                    <td
                      align="right"
                    >
                      29
                    </td>
                  </tr>
                  <tr>
                    <td>
                      <input
                        type="checkbox"
                      />
                    </td>
                    <td
                      align="right"
                    >
                      30
                    </td>
                    <td>
                      30 Guy
                    </td>
                    <td
                      align="right"
                    >
                      30
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
        <div
          style="height: 40%;"
        >
          <h2>
            Table fetching data from api
          </h2>
          <div>
            Loading...
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      id="root"
      style="padding: 10px; height: 100vh;"
    >
      <div
        style="height: 50%; overflow: hidden;"
      >
        <h2>
          With 500000 data points
        </h2>
        <div
          class="vir-table"
        >
          <div
            class="vir-table-header"
          >
            <table>
              <colgroup>
                <col
                  style="background-color: rgb(250, 250, 250); width: 20px; min-width: 20px;"
                />
                <col
                  style="background-color: rgb(250, 250, 250); width: 100px; min-width: 100px;"
                />
                <col
                  style="background-color: rgb(250, 250, 250); width: 10%; min-width: 10%;"
                />
                <col
                  style="background-color: rgb(250, 250, 250); width: 10%; min-width: 10%;"
                />
              </colgroup>
              <thead>
                <th>
                  <input
                    type="checkbox"
                  />
                </th>
                <th
                  align="right"
                >
                  ID
                </th>
                <th>
                  NAME
                </th>
                <th
                  align="right"
                >
                  IID
                </th>
              </thead>
            </table>
          </div>
          <div
            class="vir-table-body"
            style="max-height: 100%; overflow: auto;"
          >
            <table>
              <colgroup>
                <col
                  style="background-color: rgb(255, 255, 255); width: 20px; min-width: 20px;"
                />
                <col
                  style="background-color: rgb(255, 255, 255); width: 100px; min-width: 100px;"
                />
                <col
                  style="background-color: rgb(255, 255, 255); width: 10%; min-width: 10%;"
                />
                <col
                  style="background-color: rgb(255, 255, 255); width: 10%; min-width: 10%;"
                />
              </colgroup>
              <tbody>
                <tr>
                  <td>
                    <input
                      type="checkbox"
                    />
                  </td>
                  <td
                    align="right"
                  >
                    0
                  </td>
                  <td>
                    0 Guy
                  </td>
                  <td
                    align="right"
                  >
                    0
                  </td>
                </tr>
                <tr>
                  <td>
                    <input
                      type="checkbox"
                    />
                  </td>
                  <td
                    align="right"
                  >
                    1
                  </td>
                  <td>
                    1 Guy
                  </td>
                  <td
                    align="right"
                  >
                    1
                  </td>
                </tr>
                <tr>
                  <td>
                    <input
                      type="checkbox"
                    />
                  </td>
                  <td
                    align="right"
                  >
                    2
                  </td>
                  <td>
                    2 Guy
                  </td>
                  <td
                    align="right"
                  >
                    2
                  </td>
                </tr>
                <tr>
                  <td>
                    <input
                      type="checkbox"
                    />
                  </td>
                  <td
                    align="right"
                  >
                    3
                  </td>
                  <td>
                    3 Guy
                  </td>
                  <td
                    align="right"
                  >
                    3
                  </td>
                </tr>
                <tr>
                  <td>
                    <input
                      type="checkbox"
                    />
                  </td>
                  <td
                    align="right"
                  >
                    4
                  </td>
                  <td>
                    4 Guy
                  </td>
                  <td
                    align="right"
                  >
                    4
                  </td>
                </tr>
                <tr>
                  <td>
                    <input
                      type="checkbox"
                    />
                  </td>
                  <td
                    align="right"
                  >
                    5
                  </td>
                  <td>
                    5 Guy
                  </td>
                  <td
                    align="right"
                  >
                    5
                  </td>
                </tr>
                <tr>
                  <td>
                    <input
                      type="checkbox"
                    />
                  </td>
                  <td
                    align="right"
                  >
                    6
                  </td>
                  <td>
                    6 Guy
                  </td>
                  <td
                    align="right"
                  >
                    6
                  </td>
                </tr>
                <tr>
                  <td>
                    <input
                      type="checkbox"
                    />
                  </td>
                  <td
                    align="right"
                  >
                    7
                  </td>
                  <td>
                    7 Guy
                  </td>
                  <td
                    align="right"
                  >
                    7
                  </td>
                </tr>
                <tr>
                  <td>
                    <input
                      type="checkbox"
                    />
                  </td>
                  <td
                    align="right"
                  >
                    8
                  </td>
                  <td>
                    8 Guy
                  </td>
                  <td
                    align="right"
                  >
                    8
                  </td>
                </tr>
                <tr>
                  <td>
                    <input
                      type="checkbox"
                    />
                  </td>
                  <td
                    align="right"
                  >
                    9
                  </td>
                  <td>
                    9 Guy
                  </td>
                  <td
                    align="right"
                  >
                    9
                  </td>
                </tr>
                <tr>
                  <td>
                    <input
                      type="checkbox"
                    />
                  </td>
                  <td
                    align="right"
                  >
                    10
                  </td>
                  <td>
                    10 Guy
                  </td>
                  <td
                    align="right"
                  >
                    10
                  </td>
                </tr>
                <tr>
                  <td>
                    <input
                      type="checkbox"
                    />
                  </td>
                  <td
                    align="right"
                  >
                    11
                  </td>
                  <td>
                    11 Guy
                  </td>
                  <td
                    align="right"
                  >
                    11
                  </td>
                </tr>
                <tr>
                  <td>
                    <input
                      type="checkbox"
                    />
                  </td>
                  <td
                    align="right"
                  >
                    12
                  </td>
                  <td>
                    12 Guy
                  </td>
                  <td
                    align="right"
                  >
                    12
                  </td>
                </tr>
                <tr>
                  <td>
                    <input
                      type="checkbox"
                    />
                  </td>
                  <td
                    align="right"
                  >
                    13
                  </td>
                  <td>
                    13 Guy
                  </td>
                  <td
                    align="right"
                  >
                    13
                  </td>
                </tr>
                <tr>
                  <td>
                    <input
                      type="checkbox"
                    />
                  </td>
                  <td
                    align="right"
                  >
                    14
                  </td>
                  <td>
                    14 Guy
                  </td>
                  <td
                    align="right"
                  >
                    14
                  </td>
                </tr>
                <tr>
                  <td>
                    <input
                      type="checkbox"
                    />
                  </td>
                  <td
                    align="right"
                  >
                    15
                  </td>
                  <td>
                    15 Guy
                  </td>
                  <td
                    align="right"
                  >
                    15
                  </td>
                </tr>
                <tr>
                  <td>
                    <input
                      type="checkbox"
                    />
                  </td>
                  <td
                    align="right"
                  >
                    16
                  </td>
                  <td>
                    16 Guy
                  </td>
                  <td
                    align="right"
                  >
                    16
                  </td>
                </tr>
                <tr>
                  <td>
                    <input
                      type="checkbox"
                    />
                  </td>
                  <td
                    align="right"
                  >
                    17
                  </td>
                  <td>
                    17 Guy
                  </td>
                  <td
                    align="right"
                  >
                    17
                  </td>
                </tr>
                <tr>
                  <td>
                    <input
                      type="checkbox"
                    />
                  </td>
                  <td
                    align="right"
                  >
                    18
                  </td>
                  <td>
                    18 Guy
                  </td>
                  <td
                    align="right"
                  >
                    18
                  </td>
                </tr>
                <tr>
                  <td>
                    <input
                      type="checkbox"
                    />
                  </td>
                  <td
                    align="right"
                  >
                    19
                  </td>
                  <td>
                    19 Guy
                  </td>
                  <td
                    align="right"
                  >
                    19
                  </td>
                </tr>
                <tr>
                  <td>
                    <input
                      type="checkbox"
                    />
                  </td>
                  <td
                    align="right"
                  >
                    20
                  </td>
                  <td>
                    20 Guy
                  </td>
                  <td
                    align="right"
                  >
                    20
                  </td>
                </tr>
                <tr>
                  <td>
                    <input
                      type="checkbox"
                    />
                  </td>
                  <td
                    align="right"
                  >
                    21
                  </td>
                  <td>
                    21 Guy
                  </td>
                  <td
                    align="right"
                  >
                    21
                  </td>
                </tr>
                <tr>
                  <td>
                    <input
                      type="checkbox"
                    />
                  </td>
                  <td
                    align="right"
                  >
                    22
                  </td>
                  <td>
                    22 Guy
                  </td>
                  <td
                    align="right"
                  >
                    22
                  </td>
                </tr>
                <tr>
                  <td>
                    <input
                      type="checkbox"
                    />
                  </td>
                  <td
                    align="right"
                  >
                    23
                  </td>
                  <td>
                    23 Guy
                  </td>
                  <td
                    align="right"
                  >
                    23
                  </td>
                </tr>
                <tr>
                  <td>
                    <input
                      type="checkbox"
                    />
                  </td>
                  <td
                    align="right"
                  >
                    24
                  </td>
                  <td>
                    24 Guy
                  </td>
                  <td
                    align="right"
                  >
                    24
                  </td>
                </tr>
                <tr>
                  <td>
                    <input
                      type="checkbox"
                    />
                  </td>
                  <td
                    align="right"
                  >
                    25
                  </td>
                  <td>
                    25 Guy
                  </td>
                  <td
                    align="right"
                  >
                    25
                  </td>
                </tr>
                <tr>
                  <td>
                    <input
                      type="checkbox"
                    />
                  </td>
                  <td
                    align="right"
                  >
                    26
                  </td>
                  <td>
                    26 Guy
                  </td>
                  <td
                    align="right"
                  >
                    26
                  </td>
                </tr>
                <tr>
                  <td>
                    <input
                      type="checkbox"
                    />
                  </td>
                  <td
                    align="right"
                  >
                    27
                  </td>
                  <td>
                    27 Guy
                  </td>
                  <td
                    align="right"
                  >
                    27
                  </td>
                </tr>
                <tr>
                  <td>
                    <input
                      type="checkbox"
                    />
                  </td>
                  <td
                    align="right"
                  >
                    28
                  </td>
                  <td>
                    28 Guy
                  </td>
                  <td
                    align="right"
                  >
                    28
                  </td>
                </tr>
                <tr>
                  <td>
                    <input
                      type="checkbox"
                    />
                  </td>
                  <td
                    align="right"
                  >
                    29
                  </td>
                  <td>
                    29 Guy
                  </td>
                  <td
                    align="right"
                  >
                    29
                  </td>
                </tr>
                <tr>
                  <td>
                    <input
                      type="checkbox"
                    />
                  </td>
                  <td
                    align="right"
                  >
                    30
                  </td>
                  <td>
                    30 Guy
                  </td>
                  <td
                    align="right"
                  >
                    30
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
      <div
        style="height: 40%;"
      >
        <h2>
          Table fetching data from api
        </h2>
        <div>
          Loading...
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
